src/App.tsx

```diff
@@ -1,16 +1,161 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { Error } from './components/Error/Error';
+import { ErrorContext } from './components/Error/ErrorContext';
+import { Footer } from './components/Footer/Footer';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const {
+    Load,
+    Add,
+    Delete,
+    Update,
+  } = Errors;
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [title, setTitle] = useState('');
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const {
+    setHasError,
+    setCurrentError,
+    setIsAdding,
+    setSelectedTodoIds,
+  } = useContext(ErrorContext);
+
+  const loadUserTodos = useCallback(async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setHasError(true);
+        setCurrentError(Load);
+      }
+    }
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    loadUserTodos();
+  }, []);
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const todoToToggle = completedTodos.length !== todos.length
+    ? activeTodos
+    : todos;
+
+  const addNewTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    try {
+      setHasError(false);
+      setIsAdding(true);
+
+      const todo = await addTodo(todoData);
+
+      setTitle('');
+
+      setTodos(prevTodos => [...prevTodos, todo]);
+
+      setIsAdding(false);
+    } catch (error) {
+      setHasError(true);
+      setCurrentError(Add);
+    }
+  }, []);
+
+  const deleteTodo = useCallback(async (todoId: number) => {
+    try {
+      setHasError(false);
+      setSelectedTodoIds(prevIds => [...prevIds, todoId]);
+
+      await removeTodo(todoId);
+
+      await loadUserTodos();
+
+      setSelectedTodoIds([0]);
+    } catch (error) {
+      setHasError(true);
+      setCurrentError(Delete);
+    }
+  }, []);
+
+  const clearCompleted = useCallback(async () => {
+    await Promise.all(completedTodos
+      .map((todo) => deleteTodo(todo.id)));
+  }, [completedTodos]);
+
+  const updateStatus = useCallback(async (todo: Todo) => {
+    try {
+      setHasError(false);
+      setSelectedTodoIds(prevIds => [...prevIds, todo.id]);
+
+      const updatedTodo = {
+        ...todo,
+        completed: !todo.completed,
+      };
+
+      await patchTodo(todo.id, updatedTodo);
+
+      await loadUserTodos();
+
+      setSelectedTodoIds([0]);
```

Comment:
Better to move it in finally construction. So when your todo is not updated, there will be no selected todos.